### ISSUE TITLE
[Help] How to detect Ip address and get sessionID?

### ISSUE THREAD

**Issue by tubean** [fonimus/ssh-shell-spring-boot#93]
Hi @fonimus ,

Firstly thank you for this awesome library. It is very beautiful and easy to carry out.

Now I need to detect the ip address of client which request to ssh and the sessionID when connect successful. Please help me the solution.

Thank you!

**Comment by fonimus** [comment]
Hello, thanks for the feedback !

I just added a new [listener feature](https://github.com/fonimus/ssh-shell-spring-boot#listeners) which allow you to get events about start/stop sessions, with the ssh channel session.
You can then retrieve session id, remote ip, and many more.

It is in the master branch, can you test on your side if it suits your needs ? If it does I will probably make a release in the next few days.

**Comment by tubean** [comment]
Hi @fonimus ,
Thanks for your support.
I could retrieve session id, remote ip address as your guide. But when I use SshShellHelper to retrieve authentication, I got an exception "unable to find ssh context"
Here is my implementation:


```
@Component
public class SessionConfig implements SshShellListener {
  private final SshShellHelper helper;
  public SessionConfig( SshShellHelper helper) {
   this.helper = helper;
 }

@Override
public void onEvent(SshShellEvent sshShellEvent) {
  if(sshShellEvent.getType().equals(SshShellEventType.SESSION_STARTED)) {
   helper.printInfo("Connected");
  }
 }
}
```
Could you give me an advice?

**Comment by fonimus** [comment]
Hi,

I didn't think you would use the ssh helper to write on terminal directly.

I moved the listener call in the same thread as the ssh execution, it should work now.

**Comment by tubean** [comment]
Hi @fonimus ,
Actually I want to user SshShellHelper to get authentication. I found this way to get it:

`SshAuthentication authentication = (SshAuthentication) SshShellEvent.getSession().getServerSession().getIoSession().getAttribute("authentication");`

But you has made it easy when using helper.getAuthentication().

Thank you so much 💯

**Comment by fonimus** [comment]
You're welcome, I will make it available in next release : 1.3.0 later this week.
